### PR TITLE
feat: Improve form control api and docs

### PR DESCRIPTION
**Pull request by suevalov**
# Purpose of PR

* Adds default marginBottom to FormLabel, ValidationMessage, and HelpText components. Allows to override it with margin props.
* Adds different examples of using FormControl in the storybook

### CODE REVIEW

**Review by bgutsol — APPROVED** [review]

File: packages/components/forms/stories/FormControl.stories.tsx [hunk 1]
```diff
@@ -13,12 +21,74 @@ export default {
 
 export const Basic = (args: FormControlInternalProps) => {
   return (
-    <FormControl {...args}>
-      <FormControl.Label isRequired>Name</FormControl.Label>
-      <TextInput label="input" />
+    <>
+      <FormControl {...args}>
+        <FormControl.Label isRequired>Name</FormControl.Label>
+        <TextInput label="input" />
```
Comment (gui-santos) [inline packages/components/forms/stories/FormControl.stories.tsx:27]: I thought the label, in this case, would be provided by `FormControl.Label`
why do we need to pass label here?

File: packages/components/forms/stories/FormControl.stories.tsx [hunk 2]
```diff
@@ -13,12 +21,74 @@ export default {
...
+            <TextLink
+              as="button"
+              icon={<LockIcon />}
+              onClick={() => {
+                setIsDisabled(true);
+              }}
+            >
+              Lock
+            </TextLink>
+          )}
```
Comment (gui-santos) [inline packages/components/forms/stories/FormControl.stories.tsx:84]:
```suggestion
            <TextLink
              as="button"
              icon={<LockIcon />}
              onClick={() => {
                setIsDisabled(prevState => !prevState);
              }}
            >
              {isDisabled ? 'Unlock to edit' : 'Lock'}
            </TextLink>
          )}
```

I think it's possible to do the same without repeating the code as much, wdyt?

P.S.: it also teaches people about prevState

**Review by gui-santos — COMMENTED** [review]
It looks good

I only have suggestions on the stories

File: packages/components/forms/stories/FormControl.stories.tsx (hunk 1 above)
Comment (suevalov) [reply packages/components/forms/stories/FormControl.stories.tsx:27]: Right now it's enforced by TextInput API, but @burakukula is going to remove it.

File: packages/components/forms/stories/FormControl.stories.tsx (hunk 2 above)
Comment (suevalov) [reply packages/components/forms/stories/FormControl.stories.tsx:84]: yeah, it's a great suggesion

**Review by gui-santos — APPROVED** [review]

**Review by massao — APPROVED** [review]

**Pull request merged**